test/qa/lib/network.py

```diff
@@ -47,12 +61,29 @@ def _is_dns_resolvable(retry=5) -> bool:
     return False
 
 
+def _is_dns_not_resolvable(retry=5) -> bool:
+    """returns True when domain resolution is not working"""
+    for i in range(retry):
+        try:
+            with pytest.raises(sh.ErrorReturnCode_2) as ex:
+                sh.ping("-4", "-c", "1", "nordvpn.com")
+
+            return "Network is unreachable" in str(ex) or \
+                   "Name or service not known" in str(ex) or \
+                   "Temporary failure in name resolution" in str(ex)
+        except Exception as ex:
+            time.sleep(1)
+    return False
+
+
 def is_not_available(retry=5) -> bool:
-    try:
-        is_available(retry)
-        return False
-    except AssertionError:
-        return True
+    """returns True when network access is not available or throws AssertionError otherwise"""
```

Comment:
This is just my personal preference, but this could just return `True` if  network is available end `False` if it's not, and assertions could be made in tests. Or the return value could be just omitted.